Review "Backport changes" page (#498)
grafana/writers-toolkit: docs/sources/review/backport-changes/index.md

---
aliases:
  - /docs/writers-toolkit/review/backport-changes/
  - /docs/writers-toolkit/review/backporting/
  - /docs/writers-toolkit/writing-guide/tooling-and-workflows/backporting/
date: 2024-02-15
description: How and when to backport changes
keywords:
  - backporting
  - backport
menuTitle: Backport changes
title: Backport changes
weight: 500
---

# Backport changes

Projects with versioned documentation typically maintain each version with the code in multiple long-lived branches.
The `main` branch has the most recent code and documentation.

Project releases typically use long-lived branches that include the major and minor versions of the release.

> For example in the `grafana/grafana` repository, the `v9.0.x` long-lived release branch contains code and documentation for all patched versions of the major version `9` and minor version `0` of Grafana.

Reviewer: ```suggestion
For example, in the `grafana/grafana` repository, the `v9.0.x` long-lived release branch contains code and documentation for all patched versions of the major version `9` and minor version `0` of Grafana.
```

> Backporting takes a change from the `main` branch, and ports it back to another long-lived release branch.

Reviewer: ```suggestion
Backporting takes a change from the `main` branch and ports it back to another long-lived release branch.
```

Every documentation pull request to a project with versioned documentation requires a decision about whether or not to backport it.

You should backport to all supported versions of the project affected by the pull request changes.

## Backport a change

To backport a change, add the appropriate `backport <BRANCH>` label.
You can backport to more than one branch by using multiple labels.

> After the pull request is merged, Grot, the Grafana bot, creates a follow-up pull request for each of the `backport <BRANCH>` labels.

Reviewer: ```suggestion
After you merge the pull request, Grot, the Grafana bot, creates a follow-up pull request for each of the `backport <BRANCH>` labels.
```

> If Grot is unable to automatically backport the changes, it comments on the first pull request with instructions about how to backport the change manually.

Reviewer: ```suggestion
If Grot can't automatically backport the changes, it comments on the first pull request with instructions about how to backport the change manually.
```

In repositories such as `grafana/grafana`, engineers sometimes create a branch for a release well before the release has shipped.

> If you intend to publish content against an imminent release, check for the existence of a backport label for the upcoming version before merging the pull request.

Reviewer: ```suggestion
If you intend to publish content against an imminent release, check for a backport label for the upcoming version before merging the pull request.
```

If you decide to _not_ backport a change, you don't need to add any label.

## When to backport

To decide whether to backport a pull request, use the following decision tree:

<!-- vale Grafana.Timeless = NO -->

<script type="module">
  import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';
  mermaid.initialize({ startOnLoad: true });
</script>
<div class="mermaid">
  flowchart TD
  D1{Is the change documenting a new feature?}
  D2{Is the feature going to be released in<br>a future version of the project that<br>doesn't yet have a release branch?}
  D3{Is the change a fix for a typo?}
  D4{Is the change a documentation refactoring?}
  T1[Do nothing.]
  T2[Add a backport label for each affected release version.]
  T3[Update this flowchart as the decision making process is incomplete.]
  T4[Backport the change to release branch for the next version of the project.]
  D1 -- Yes --> D2
  D1 -- No -->  D3
  D2 -- Yes --> T1
  D2 -- No -->  T4
  D3 -- Yes --> T2
  D3 -- No -->  D4
  D4 -- Yes --> T4
  D4 -- No -->  T3
</div>

<!-- vale Grafana.Timeless = YES -->

## Backporting tutorials

> Depending on the location from which the website mounts tutorials content, you might need to backport changes to them.

Reviewer: ```suggestion
Depending on the location from which the website mounts tutorial content, you might need to backport changes to them.
```

> For `grafana/grafana`, the website mounts tutorials content from the `next` version of documentation, so you don't need to backport changes.

Reviewer: ```suggestion
For `grafana/grafana`, the website mounts tutorial content from the `next` version of the documentation, so you don't need to backport changes.
```
